Figgle.Generator/EmbedFontSourceGenerator.cs

```diff
@@ -0,0 +1,329 @@
+﻿// Copyright Drew Noakes. Licensed under the Apache-2.0 license. See the LICENSE file for more details.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Figgle.Fonts;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Figgle.Generator;
+
+[Generator(LanguageNames.CSharp)]
+internal sealed class EmbedFontSourceGenerator : IIncrementalGenerator
```

Comment:
Use of `IIncrementalGenerator` is nice here! I've wondered how difficult it'd be to convert `FiggleSourceGenerator` to be incremental. Not something for this PR, but I've never made an incremental generator before so am curious what your thoughts are. Given the generator does a bunch of file IO to read/parse a font, I wonder if incrementality would be good there too.